smt.go

```diff
@@ -340,11 +344,15 @@ func (smt *SparseMerkleTree) sideNodesForRoot(path []byte, root []byte) ([][]byt
 		leftNode, rightNode := smt.th.parseNode(currentData)
 
 		// Get sidenode depending on whether the path bit is on or off.
-		if hasBit(path, i) == right {
-			sideNodes[i] = leftNode
+		if getBitAtFromMSB(path, i) == right {
+			sideNodes = append(sideNodes, nil)
+			copy(sideNodes[1:], sideNodes)
+			sideNodes[0] = leftNode
```

Comment:
(Wow, it takes a lot of time (at least for me) to fully understand why it works as it should.)
I would say below express the intention better:
`sideNodes = append([][]byte{leftNode}, sideNodes...)`

Reply:
Apparently the current implementation is more efficient since it does fewer allocations: https://stackoverflow.com/a/53737602 (assuming [there's actually space in the slice](https://github.com/lazyledger/smt/pull/22#discussion_r590732071)).